perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -1465,22 +960,40 @@ def Run(benchmark_spec):
                'scenario is %s, storage provider is %s.',
                FLAGS.object_storage_scenario, FLAGS.storage)
 
+  service = benchmark_spec.service
+  buckets = benchmark_spec.buckets
+
   metadata = {'storage provider': FLAGS.storage}
 
   vms = benchmark_spec.vms
 
-  # The client tool based tests requires some provisioning on the VMs first.
-  vms[0].RemoteCommand(
-      'cd %s/run/; bash cloud-storage-workload.sh %s' % (vms[0].GetScratchDir(),
-                                                         FLAGS.cli_test_size))
   if FLAGS[OBJECT_STORAGE_REGION].present:
     metadata[REGIONAL_BUCKET_LOCATION] = FLAGS.object_storage_region
   else:
     metadata[REGIONAL_BUCKET_LOCATION] = DEFAULT
 
-  results = OBJECT_STORAGE_BENCHMARK_DICTIONARY[FLAGS.storage].Run(vms[0],
-                                                                   metadata)
-  print results
+  if FLAGS[OBJECT_STORAGE_GCS_MULTIREGION].present:
+    metadata[GCS_MULTIREGION_LOCATION] = FLAGS.object_storage_gcs_multiregion
+  else:
+    metadata[GCS_MULTIREGION_LOCATION] = DEFAULT
+
+  metadata.update(service.Metadata(vms[0]))
+
+  results = []
+  test_script_path = '%s/run/%s' % (vms[0].GetScratchDir(), API_TEST_SCRIPT)
+  command_builder = APIScriptCommandBuilder(
+      test_script_path, STORAGE_TO_API_SCRIPT_DICT[FLAGS.storage], service)
```

Comment:
Fixed. I also made sure that PKB would print a useful error message if the user tries to run API benchmarks on Swift.